knimin/lib/data_access.py

```diff
@@ -2399,6 +2399,178 @@ def getProjectNames(self):
+
+        Parameters
+        ----------
+        sample_ids : list of str
+            Sample IDs to delete
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """DELETE FROM pm.sample WHERE sample_id = %s"""
+        self._con.executemany(sql, [[x] for x in sample_ids])
+        return True
+
+    def create_sample_plate(self, sample_plate_info):
+        """Creates a new sample plate and sets values for mandatory fields
+
+        Parameters
+        ----------
+        sample_plate_info: tuple of (str, str, datetime, str, str)
+            name, email, creation_timestamp, notes, plate_type
+
+        Returns
+        -------
+        int
+            ID of the created sample plate
+        """
+        sql = """INSERT INTO pm.sample_plate (name, email, creation_timestamp,
+                 notes, plate_type_id) VALUES (%s, %s, %s, %s,
+                 (SELECT plate_type_id FROM pm.plate_type WHERE name = %s))
+                 RETURNING sample_plate_id"""
+        sql_args = list(sample_plate_info)
+        for i in range(1, 4):
+            sql_args[i] = sql_args[i] or None
+        return self._con.execute_fetchone(sql, sql_args)[0]
+
+    def set_sample_plate_info(self, sample_plate_id, sample_plate_info):
+        """Sets attributes of a sample plate by ID
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            ID of the sample plate to set attributes to
+        sample_plate_info: tuple of (str x2, datetime, str x2)
+            name, email, creation_timestamp, notes, plate_type
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """UPDATE pm.sample_plate
+                 SET name = %s, email = %s, creation_timestamp = %s,
+                    notes = %s, plate_type_id = (SELECT plate_type_id
+                    FROM pm.plate_type WHERE name = %s)
+                 WHERE sample_plate_id = %s"""
+        sql_args = list(sample_plate_info + (sample_plate_id,))
+        for i in range(1, 4):
+            sql_args[i] = sql_args[i] or None
+        self._con.execute(sql, sql_args)
+        return True
+
+    def get_sample_plate_info(self, sample_plate_id):
+        """Gets attributes of a sample plate by ID
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            ID of the sample plate to get attributes from
+
+        Returns
+        -------
+        tuple of (str x2, datetime, str x2)
+            name, email, creation_timestamp, notes, plate_type
+        """
+        sql = """SELECT p.name, p.email, p.creation_timestamp, p.notes,
+                    pm.plate_type.name
+                 FROM pm.sample_plate p
+                 JOIN pm.plate_type USING (plate_type_id)
+                 WHERE sample_plate_id = %s"""
+        return tuple(self._con.execute_fetchone(sql, [sample_plate_id]))
+
+    def delete_sample_plate(self, sample_plate_ids):
```

Comment:
Change to `delete_sample_plates`.

In which case are we going to delete multiple sample plates at once? I think it is safer if we only allow to remove 1 sample plate at a time (note that once it is removed is gone forever, so we want to protect a bit better for accidental deletions).

Also, this will fail in case that the SamplePlate is associated with a DNA plate.